### ISSUE TITLE
Support half-highlighted text by overloading gradient color stops?

### ISSUE THREAD

**Issue by DoctorDerek** [tailwindlabs/tailwindcss#3954]
### What version of @tailwindcss/jit are you using?

Latest (v0.1.7)

### What version of Node.js are you using?

v14.15.1

### What browser are you using?

Chrome latest 64-bit 89.0.etc

### What operating system are you using?

Windows 10

### Reproduction repository

https://codepen.io/doctorderek/pen/eYBaPRv?editors=0100

![image](https://user-images.githubusercontent.com/761231/112228746-c8fedf80-8bff-11eb-9dfb-bd1315ba8c1c.png)

I like the "half-highlight" style for links that you'll see occasionally. Here's the CSS I want to duplicate in Tailwind:
```
background: linear-gradient(to bottom, rgba(41,148,178,0), rgba(41,148,178,0) 50%,rgba(41,148,178,1) 50%, rgba(41,148,178,1) 100%);
```

I'm happy to implement the code for this feature, but I don't even know what the Tailwind syntax would be.

I was thinking there could be a way to do this using Tailwind JIT, but it would require extending the syntax.

### Other examples

This is not the same, because I need to overload the "via" selector and have two selectors:
```
<h1 class="bg-gradient-to-t from-blue-500 via-blue-500 to-transparent">Garlic</h1>
```
![image](https://user-images.githubusercontent.com/761231/112229125-63f7b980-8c00-11eb-8155-946403895c6e.png)

Nor is this:
```
<h1 class="bg-gradient-to-t from-blue-500 via-transparent to-transparent">Garlic</h1>
```
![image](https://user-images.githubusercontent.com/761231/112229300-bafd8e80-8c00-11eb-939a-855433f55943.png)

Basically the issue is that there's no "gradient override" in the JIT system, so I currently need custom CSS.

**Comment by paulm17** [comment]
I have something similar to you.  So adding it here instead of making a new issue.

```
.cpHueHorizontal {
  background: linear-gradient(
    to right,
    hsl(0, 100%, 50%),
    hsl(60, 100%, 50%),
    hsl(120, 100%, 50%),
    hsl(180, 100%, 50%),
    hsl(240, 100%, 50%),
    hsl(300, 100%, 50%),
    hsl(360, 100%, 50%)
  );
}
.cpHueVertical {
  background: linear-gradient(
    to bottom,
    hsl(0, 100%, 50%),
    hsl(60, 100%, 50%),
    hsl(120, 100%, 50%),
    hsl(180, 100%, 50%),
    hsl(240, 100%, 50%),
    hsl(300, 100%, 50%),
    hsl(360, 100%, 50%)
  );
}
```

It would be awesome to do something like this:

```
bg-gradient-to-b gradient-stops-[#FF0000, #FFFF00, #00FF00, #00FFFF, #0000FF, #FF00FF, #FF0004]
```

**Comment by bradlc** [comment]
Hey @DoctorDerek. You can create this effect by extending the `backgroundImage` theme config. Like this: https://play.tailwindcss.com/QvgPyqyia8?file=config
